crates/bitwarden-crypto/src/keys/master_key.rs

```diff
@@ -118,39 +124,44 @@ impl MasterKey {
     }
 
     pub fn to_base64(&self) -> String {
-        self.0.to_base64()
+        STANDARD.encode(self.0.key_material.to_vec())
+    }
+
+    pub fn try_from(value: &mut [u8]) -> Result<Self> {
+        let material = KdfDerivedKeymaterial {
+            key_material: Box::pin(GenericArray::<u8, U32>::clone_from_slice(value)),
+        };
+        value.zeroize();
+        Ok(Self::new(material))
     }
 }
 
 /// Helper function to encrypt a user key with a master or pin key.
 pub(super) fn encrypt_user_key(
-    key: &SymmetricCryptoKey,
+    key: &KdfDerivedKeymaterial,
     user_key: &SymmetricCryptoKey,
 ) -> Result<EncString> {
     let stretched_key = stretch_kdf_key(key)?;
-
-    EncString::encrypt_aes256_hmac(
-        &user_key.to_vec(),
-        stretched_key
-            .mac_key
-            .as_ref()
-            .ok_or(CryptoError::InvalidMac)?,
-        &stretched_key.key,
-    )
+    EncString::encrypt_aes256_hmac(&user_key.to_vec(), &stretched_key)
```

Comment:
To_vec was used here before my change, does this need to be zeroized?

Reply:
Added zeroizing now.

Reply:
As a general rule, anything that is heap allocated will be zeroized on drop by our custom allocator, so `Strings`, `Vecs`, etc, will be handled automatically for you. Even though we don't support it, users can technically disable this custom allocator with the `no-memory-hardening` feature in case they need a different one or it produces incompatibilities, so it doesn't hurt to explicitly call zeroize, but in this case it shouldn't be necessary.

The one case where we always need to zeroize manually is for stack allocated arrays, for example if you're using `[u8; 32]` or `GenericArray<>`. In those cases you either need to wrap them in a `Box` to move them to the heap or ensure you zeroize it every time.